tests/queries/0_stateless/01455_leaf_max_rows_bytes_to_read.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+. "$CURDIR"/../shell_config.sh
+
+$CLICKHOUSE_CLIENT --query="DROP TABLE IF EXISTS test_local;"
+$CLICKHOUSE_CLIENT --query="DROP TABLE IF EXISTS test_distributed;"
+
+$CLICKHOUSE_CLIENT --query="CREATE TABLE test_local (date Date, value UInt32) ENGINE = MergeTree(date, date, 8192);"
+$CLICKHOUSE_CLIENT --query="CREATE TABLE test_distributed AS test_local ENGINE = Distributed(test_shard_localhost, currentDatabase(), test_local, rand());"
+$CLICKHOUSE_CLIENT --query="INSERT INTO test_local SELECT '2000-08-01', number as value from numbers(50000);"
+
+# Must throw an exception
+EXCEPTION_TEXT="Limit for rows (controlled by 'max_rows_to_read_leaf' setting) exceeded"
+$CLICKHOUSE_CLIENT --query="SELECT count() FROM (SELECT * FROM test_distributed) SETTINGS max_rows_to_read_leaf = 40000;" 2>&1 \
+    | grep -q "$EXCEPTION_TEXT" && echo "$EXCEPTION_SUCCESS_TEXT" || echo "Did not thrown an exception"
```

Comment:
Why not simply `select count() from test+distributed settings ...` ?

Reply:
`select count() from test_distributed settings ...` doesn't check these limits due to some optimisations, I believe. For example, it won't fail for `max_rows_to_read`:
```
SELECT count()
FROM test_distributed
SETTINGS max_rows_to_read = 40000

┌─count()─┐
│  200000 │
└─────────┘

1 rows in set. Elapsed: 0.024 sec. 
```

Reply:
Oh, I see, this is because of `optimize_trivial_count_query`